src/photos/ducks/clustering/matching.js

```diff
@@ -40,9 +48,10 @@ export const matchingClusters = (photo, albums) => {
     return [edge]
   }
   for (let i = 0; i < albums.length; i++) {
-    const matches = adjacentToClusters(photo, albums[i], albums[i + 1])
-    if (matches.length > 0) {
-      return matches
+    if (photoInsideCluster(photo, albums[i])) {
+      return [albums[i]]
+    } else if (photoBetweenClusters(photo, albums[i], albums[i + 1])) {
```

Comment:
Yep, I think this is better! Now we could even *not* call `photoBetweenClusters ` when there is no `albums[i + 1]` — that way the `photoBetweenClusters` function doesn't have to check it's parameter.